java/core/src/test/java/com/google/protobuf/GeneratedMessageTest.java

```diff
@@ -853,6 +854,30 @@ public void testSerializePartial() throws Exception {
     assertEquals(expected, actual);
   }
 
+  public void testSerializeBackwardsCompatibility() throws Exception {
```

Comment:
Mention somewhere what "version" of the class this is. Namely, that it was before the introduction of the `Class`.